pkg/apis/query/v0alpha1/query.go

```diff
@@ -75,6 +80,21 @@ type GenericDataQuery struct {
 	props map[string]any `json:"-"`
 }
 
+type ResultDataContract struct {
+	// Type asserts that the frame matches a known type structure.
+	Type data.FrameType `json:"type,omitempty"`
+
+	// TypeVersion is the version of the Type property. Versions greater than 0.0 correspond to the dataplane
+	// contract documentation https://grafana.github.io/dataplane/contract/.
+	TypeVersion data.FrameTypeVersion `json:"typeVersion"`
+
```

Comment:
I like the limits, I but I think maybe they belong in another property - then maybe naming is easier (e.g. (Limits/ExpectedType or something)?

Reply:
yes -- but nervous about adding too many properties at the root, given that we need to pick names that do not collide with anything in the wild 🤦🏻